lib/cli/run-compose.js

```diff
@@ -52,83 +57,145 @@ module.exports = async () => {
+      promptMessage += ' A "package.json" file will also be created in your current directory.';
+    }
 
-    cliProgressFooter.updateProgress('Installing Serverless Compose CLI');
+    const shouldInstallCompose = (
+      await inquirer.prompt({
+        message: promptMessage,
+        type: 'confirm',
+        name: 'shouldInstallCompose',
+      })
+    ).shouldInstallCompose;
+
+    // Add progress bar
+    if (shouldInstallCompose) {
+      const getCliProgressFooter = require('cli-progress-footer');
+      const cliProgressFooter = getCliProgressFooter();
+      cliProgressFooter.shouldAddProgressAnimationPrefix = true;
+      cliProgressFooter.progressAnimationPrefixFrames =
+        cliProgressFooter.progressAnimationPrefixFrames.map((frame) => `\x1b[91m${frame}\x1b[39m`);
+
+      cliProgressFooter.updateProgress('Installing Serverless Compose CLI');
+
+      try {
+        if (!packageJsonContent) {
+          try {
+            await ensureMinimalPackageJson();
+          } catch {
+            process.stdout.write(
+              `${[
+                '',
+                'Could not create "package.json" in current directory.',
+                'Please create it manually and run this command again.',
+              ].join('\n')}\n`
+            );
+            return;
+          }
+        }
 
-    try {
-      if (!hasPackageJson) {
         try {
-          await ensureMinimalPackageJson();
+          await spawn('npm', ['install', '--save-dev', '@serverless/compose']);
         } catch {
           process.stdout.write(
             `${[
               '',
-              'Could not create "package.json" in current directory.',
-              'Please create it manually and run this command again.',
+              'Could not install Serverless Compose CLI locally.',
+              'Please install it manually with "npm i --save-dev @serverless/compose" and run this command again.',
             ].join('\n')}\n`
           );
           return;
         }
+      } finally {
+        cliProgressFooter.updateProgress();
       }
+      hasInstalledCompose = true;
+    } else {
+      process.stdout.write(
+        `${[
+          '',
+          'Please install it manually with "npm i --save-dev @serverless/compose" and run this command again.',
+        ].join('\n')}\n`
+      );
+      return;
+    }
+  } else {
+    // Non-interactive scenario
+
+    // Here, we want to check if user has `@serverless/compose` in `devDependencies` of `package.json`
+    // If that is the case, we want to run `npm install` to ensure it's installed
+    // In all other scenarios, we want to inform user that `@serverless/compose` needs to be installed first (and/or added to "package.json")
+    process.stdout.write(
+      `${[
+        '',
+        'Serverless Compose CLI installation not found. Attempting to install it, if it\'s listed in "package.json" of your project.',
+      ].join('\n')}\n`
+    );
+
+    const composeVersionToInstall = _.get(
```

Comment:
I think there's no need to do that. `npm install` always reads local package.json and ensures to install the version from the range listed in [dev]dependencies.

Reply:
thanks - fixed, along with a small change to the messages printed to users